mobile/native/src/trade/order/handler.rs

```diff
@@ -84,40 +83,33 @@ pub async fn submit_order(
         });
     }
 
-    let url = format!("http://{}", config::get_http_endpoint());
-    let url = Url::parse(&url).expect("correct URL");
-    let orderbook_client = OrderbookClient::new(url);
-
     db::insert_order(order.clone()).map_err(SubmitOrderError::Storage)?;
 
     let runtime = get_or_create_tokio_runtime().expect("to be ablet to get runtime");
-    if let Err(err) = runtime
+    runtime
         .spawn_blocking({
-            let order = order.clone();
             move || {
-                let mut db = db::connection()?;
-                // We need to ensure that the channel-opening parameters are inserted after the
-                // order has been posted, but before order has been match and we need them to open
-                // the DLC channel. Without this database transaction we can easily end up with a
-                // match before we insert them!
-                db.transaction(|conn| {
-                    if let Some(channel_opening_params) = channel_opening_params {
-                        tracing::debug!(
-                            ?channel_opening_params,
-                            "Recording channel-opening parameters"
-                        );
-
-                        db::insert_channel_opening_params(conn, channel_opening_params)
-                            .map_err(SubmitOrderError::Storage)?;
-                    }
-
-                    runtime.block_on(orderbook_client.post_new_order(order.into()))
-                })
+                let mut conn = db::connection()?;
```

Comment:
🔧 Should go within the `if let Some` block.

Reply:
Wouldn't we swallow the error with a `if let Some` ?